File: src/main/resources/static/js/app/UserService.js
'use strict'

angular.module('devopsDemo.services', []).factory('UserService',
		[ "$http", "CONSTANTS", function($http, CONSTANTS) {
			var service = {};
			service.getUserById = function(userId) {
				var url = CONSTANTS.getUserByIdUrl + userId;
				return $http.get(url);
			}
			service.getAllUsers = function() {
				return $http.get(CONSTANTS.getAllUsers);
			}
			service.getCircleData = function() {
				return $http.get(CONSTANTS.getCircleChartData);
			}
			service.saveUser = function(userDto) {
				return $http.post(CONSTANTS.saveUser, userDto);
			}
			return service;
		} ]);

File: src/main/java/com/neilo/aithre/service/UserService.java
package com.neilo.aithre.service;

import java.util.List;

import com.neilo.aithre.dto.UserDto;

public interface UserService {
    UserDto getUserById(Integer userId);
    void saveUser(UserDto userDto);
    List<UserDto> getAllUsers();
	List<UserDto> getAllUsersBySkillName(String skillName);
	List<UserDto> getAllUsersBySkillId(Integer skillId);
}
